platform/darwin/test/MGLFilterTests.mm

```diff
@@ -20,7 +20,6 @@ - (void)setUp
     source = [[MGLGeoJSONSource alloc] initWithSourceIdentifier:@"test-source" geoJSONData:geoJSONData];
     [self.mapView.style addSource:source];
     layer = [[MGLLineStyleLayer alloc] initWithLayerIdentifier:@"test-layer" sourceIdentifier:@"test-source"];
-    [self.mapView.style addLayer:layer];
```

Comment:
@incanus I just realized that the reason the simple test was breaking is that, since the `tearDown` method calls `removeLayer` and since that code path throws an exception if you attempt to remove a layer that has not been added, adding any test that does not add the tests `layer` ivar will cause the error we saw. I think we can just chalk this up to shared setup and cleanup methods sometimes causing issues like this. In any case, breaking the tests apart was the way to go here.